examples/helia-script-tag/src/createLibp2pInstance.js

```diff
@@ -0,0 +1,89 @@
+/** global Helia, Libp2P, ChainsafeLibp2PYamux, Libp2PWebsockets, Libp2PBootstrap, BlockstoreCore, DatastoreCore, Libp2PKadDht, Libp2PMplex */
+// not imported from skypack,jsdelivr, nor unpkg because of issues. see https://github.com/ChainSafe/js-libp2p-noise/issues/296
+import { noise } from 'https://esm.sh/v111/@chainsafe/libp2p-noise@11.0.1/es2022/libp2p-noise.js'
+
+/**
+ *
+ * @param {object} param0
+ * @param {import('libp2p').Libp2pOptions['datastore']} param0.datastore
+ */
+export async function createLibp2pInstance ({datastore}) {
+  /** @type {Partial<import('libp2p').Libp2pOptions>} */
+  const libp2pInit = {
+    /**
+     * @see https://github.com/ipfs/js-ipfs-interfaces/tree/master/packages/interface-datastore
+     */
+    datastore,
+    /**
+     * @see https://github.com/libp2p/js-libp2p-kad-dht
+     * @see https://github.com/libp2p/js-libp2p/blob/master/doc/CONFIGURATION.md#customizing-dht
+     */
+    dht: /** @type {import('libp2p').Libp2pOptions['dht']} */(Libp2PKadDht.kadDHT({                        // The DHT options (and defaults) can be found in its documentation
+      kBucketSize: 5,
+      clientMode: true,
+      enabled: true,
+      randomWalk: {
+        enabled: true,            // Allows to disable discovery (enabled by default)
+        interval: 300e3,
+        timeout: 10e3
+      }
+    })),
+    /**
+     * @see https://github.com/libp2p/js-libp2p/blob/master/doc/CONFIGURATION.md#transport
+     */
+    transports: /** @type {import('libp2p').Libp2pOptions['transports']} */([
+      Libp2PWebsockets.webSockets(),
+    ]),
+    /**
+     * @see https://github.com/libp2p/js-libp2p/blob/master/doc/CONFIGURATION.md#connection-encryption
+     */
+    connectionEncryption: /** @type {import('libp2p').Libp2pOptions['connectionEncryption']} */([
+      noise()
+    ]),
+    /**
+     * @see https://github.com/libp2p/js-libp2p/blob/master/doc/CONFIGURATION.md#stream-multiplexing
+     */
+    streamMuxers: /** @type {import('libp2p').Libp2pOptions['streamMuxers']} */([
+      ChainsafeLibp2PYamux.yamux(),
+      Libp2PMplex.mplex(),
```

Comment:
Can we remove this?  I'm not aware of any reason to perpetuate mplex.

Reply:
unfortunately, I discovered that we can't talk to web3 bootstrap node without it, but I can remove it from this example

Reply:
see https://github.com/SgtPooki/helia-service-worker-gateway/blob/d7528f3c2293a47e708213340dd92b94d54109a7/src/getLibp2p.ts#L35

Reply:
also just posted in web3-storage slack: https://filecoinproject.slack.com/archives/C027XP5BTGB/p1679957587153879

Reply:
Here is the relevant Elastic iPFS issue to track: https://github.com/elastic-ipfs/elastic-ipfs/issues/25